AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/Support/MockSQLiteStorageEngineAdapter.swift

```diff
@@ -0,0 +1,190 @@
+
+    var resultForQuery: DataStoreResult<[Model]>?
+    var resultForSave: DataStoreResult<Model>?
+
+    var resultForQueryMutationSyncMetadata: MutationSyncMetadata?
+    var errorToThrowOnMutationSyncMetadata: DataStoreError?
+
+    var shouldReturnErrorOnSaveMetadata: Bool
+    var shouldReturnErrorOnDeleteMutation: Bool
+
+    var resultForQueryModelSyncMetadata: ModelSyncMetadata?
+    var listenerForModelSyncMetadata: BasicClosure?
+
+    let testCase: XCTestCase
+
+    init(testCase: XCTestCase) {
+        self.testCase = testCase
+        self.shouldReturnErrorOnSaveMetadata = false
+        self.shouldReturnErrorOnDeleteMutation = false
+    }
+
+    func setUp(models: [Model.Type]) throws {
+        testCase.recordFailure(withDescription: "Not expected to execute",
+                               inFile: #file,
+                               atLine: #line,
+                               expected: true)
+    }
+
+    // MARK: - Responses
+
+    func returnOnQuery(dataStoreResult: DataStoreResult<[Model]>?) {
+        resultForQuery = dataStoreResult
+    }
+
+    func returnOnQueryMutationSyncMetadata(_ mutationSyncMetadata: MutationSyncMetadata?) {
+        resultForQueryMutationSyncMetadata = mutationSyncMetadata
+    }
+
+    func returnOnSave(dataStoreResult: DataStoreResult<Model>?) {
+        resultForSave = dataStoreResult
+    }
+
+    func returnOnQueryModelSyncMetadata(_ metadata: ModelSyncMetadata?, listener: BasicClosure? = nil) {
+        resultForQueryModelSyncMetadata = metadata
+        listenerForModelSyncMetadata = listener
+    }
+
+    func throwOnQueryMutationSyncMetadata(error: DataStoreError) {
+        errorToThrowOnMutationSyncMetadata = error
+    }
+
+    // MARK: - StorageEngineAdapter
+
+    func delete<M: Model>(_ modelType: M.Type,
+                          withId id: Model.Identifier,
+                          completion: DataStoreCallback<Void>) {
+        testCase.recordFailure(withDescription: "Not expected to execute",
+        inFile: #file,
+        atLine: #line,
+        expected: true)
+    }
+
+    func delete(untypedModelType modelType: Model.Type,
+                withId id: String,
+                completion: (Result<Void, DataStoreError>) -> Void) {
+        return shouldReturnErrorOnDeleteMutation
+            ? completion(.failure(causedBy: DataStoreError.invalidModelName("DelMutate")))
+            : completion(.emptyResult)
+    }
+
+    func query(untypedModel modelType: Model.Type,
+               predicate: QueryPredicate?,
+               completion: DataStoreCallback<[Model]>) {
+        let result = resultForQuery ?? .failure(DataStoreError.invalidOperation(causedBy: nil))
+        completion(result)
+    }
+
+    func query<M: Model>(_ modelType: M.Type, predicate: QueryPredicate?, completion: DataStoreCallback<[M]>) {
+        testCase.recordFailure(withDescription: "Not expected to execute", inFile: #file, atLine: #line, expected: true)
```

Comment:
Why did we change over to using `recordFailure` & passing in a reference to the test case vs. what we had before: 
```
        XCTFail("not expected to execute")
```

I tried to look for a difference but I execute a test case, the console output looks exactly the same between the two.

Reply:
I had originally put that there because I needed an expectation, so I was doing `testCase.expecation(...)`. I stopped using that at some point, and never reverted.

Changed back to straight-up XCTFail.